### PR TITLE
Provide a single Terraform configuration (WIP)

### PR DESCRIPTION
**Pull request by lucperkins**
At the moment, we're maintaining separate Terraform configurations for each messaging platform. There's a lot of redundancy here that's easy to eliminate.

### CODE REVIEW

File: deploy/aws/variables.tf [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  }
+}
+
+variable "num_instances" {
+  type = "map"
+
+  default = {
+    "client"    = 4
+    "messaging" = 3
+    "zookeeper" = 0
```
Comment (merlimat) [inline deploy/aws/variables.tf]: @lucperkins is this intended to be overrided by specific terraform configs?

Comment (lucperkins) [reply deploy/aws/variables.tf]: Yes, each platform is intended to have its own `.tfvars` file (`pulsar.tfvars`, `kafka.tfvars`, etc.)

**Comment by vongosling** [comment]
@Hellojungle

**Comment by Hellojungle** [comment]
Maybe providing more cloud platforms terraform configurations?

**Pull request closed without merging**